File: ta-sipta-mariaine/app/Models/Dosen.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Dosen extends Model
{
    protected $table = 'dosen';

    protected $fillable = [
        'user_id',
        'nama_dosen',
        'nip',
        'tempat_lahir',
        'tanggal_lahir',
        'jenis_kelamin',
        'jabatan',
        'program_studi_id',
        'ttd_dosen',
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function programStudi()
    {
        return $this->belongsTo(ProgramStudi::class, 'program_studi_id');
    }

    public function pembimbingUtama()
    {
        return $this->belongsTo(Dosen::class, 'pembimbing_utama_id');
    }

    public function pembimbingPendamping()
    {
        return $this->belongsTo(Dosen::class, 'pembimbing_pendamping_id');
    }

    public function jadwalBimbingan()
    {
        return $this->hasMany(JadwalBimbingan::class);
    }
}

File: sso-auth/models/dosen.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Dosen struct {
	ID             uint           `gorm:"primaryKey" json:"id"`
	UserID         uint           `gorm:"uniqueIndex;not null" json:"user_id"`
	NIDN           string         `gorm:"uniqueIndex" json:"nidn"`
	NIP            string         `gorm:"uniqueIndex" json:"nip"`
	TempatLahir    string         `json:"tempat_lahir"`
	TanggalLahir   *time.Time     `json:"tanggal_lahir"`
	JenisKelamin   string         `json:"jenis_kelamin"`
	ProgramStudiID *uint          `json:"program_studi_id"`
	Jabatan        string         `json:"jabatan"`
	NoTelepon      string         `json:"no_telepon"`
	Alamat         string         `gorm:"type:text" json:"alamat"`
	CreatedAt      time.Time      `json:"created_at"`
	UpdatedAt      time.Time      `json:"updated_at"`
	DeletedAt      gorm.DeletedAt `gorm:"index" json:"-"`

	User         User          `gorm:"foreignKey:UserID" json:"user,omitempty"`
	ProgramStudi *ProgramStudi `gorm:"foreignKey:ProgramStudiID" json:"program_studi,omitempty"`
}
